subsys/bluetooth/audio/has.c

```diff
@@ -1200,6 +1289,61 @@ int bt_has_register(const struct bt_has_register_param *param)
 		has.features |= BT_HAS_FEAT_WRITABLE_PRESETS_SUPP;
 	}
 
+	return 0;
+}
+
+int bt_has_features_set(const struct bt_has_features_register_param *features)
+{
+	int err;
+
+	/* Check whether any features will change, otherwise we don't want to notify clients */
+	if (has.features != features->type ||
+	    (has.features & BT_HAS_FEAT_PRESET_SYNC_SUPP) ^ features->preset_sync_support ||
+	    (has.features & BT_HAS_FEAT_INDEPENDENT_PRESETS) ^ features->independent_presets) {
+		return 0;
+	}
+
+	err = has_features_register(features);
+	if (err != 0) {
+		LOG_DBG("Failed to register features");
+		return err;
+	}
+
+#if defined(CONFIG_BT_HAS_FEATURES_NOTIFIABLE)
+	for (size_t i = 0U; i < ARRAY_SIZE(has_client_list); i++) {
+		struct has_client *client = &has_client_list[i];
+			/* mark to notify */
+			client->pending_ntf.features = true;
+	}
+
+	k_work_submit(&features_work);
```

Comment:
What if this fails?

Reply:
Hmm, looking at the API description, return value greater than or equal to 0 can be considered as success, while less than 0 can mean that there is a system work queue issue that we cannot work-around here actually, so if you want to handle failing case here, I think the only possibility is to assert.

Reply:
Or reject the request to change the features, similarly if something else went wrong, right? We can propagate the error to the caller

Reply:
> Or reject the request to change the features, similarly if something else went wrong, right? We can propagate the error to the caller

I agree with this

Reply:
Similar to what we have discussed with PACS; if this fails, we should revert any changes as well

Reply:
> Does it actually makes sense to also add this into the shell app?

Not sure what you mean here - Can you elaborate? 

> And if so also expose an features_get API, and add that too?

If that has any use for anyone, then sure - But maybe add that in a separate PR

Reply:
> Not sure what you mean here - Can you elaborate?
I'm wondering if it would make sense to add a shell command which sets the feature bits runtime.